app/importers/screening_list/make_name_variants.rb

```diff
@@ -12,45 +12,55 @@ module MakeNameVariants
-        unless (doc[:alt_names_idx].map(&:downcase).join(' ').split & common_words).empty?
-          doc[:alt_names_with_common]    = doc[:alt_names_idx]
-          doc[:rev_alt_with_common]      = doc[:alt_names_with_common]
-          doc[:trim_alt_with_common]     = doc[:alt_names_with_common].map { |name| name.gsub(/\s+/, '') }
-          doc[:trim_rev_alt_with_common] = doc[:rev_alt_with_common].map { |name| name.gsub(/\s+/, '') }
-          doc[:alt_names_idx]            = doc[:alt_names_idx].map { |name| name.split.delete_if { |word| common_words.include?(word.downcase) }.join(' ') }
-        end
-
-        doc[:rev_alt_names]      = doc[:alt_names_idx].map { |name| name.split.reverse.join(' ') }
-        doc[:trim_alt_names]     = doc[:alt_names_idx].map { |name| name.gsub(/\s+/, '') }
-        doc[:trim_rev_alt_names] = doc[:rev_alt_names].map { |name| name.gsub(/\s+/, '') }
-      end
-      doc
+      doc[:name_idx] = strip_punct(doc[:name])
+      doc[:name_idx] = filter_words(doc[:name_idx], STOPWORDS)
+
+      make_names_with_common(doc, 'name') unless (doc[:name_idx].downcase.split & COMMON_WORDS).empty?
+
+      doc[:name_rev]      = name_rev(doc[:name_idx])
+      doc[:name_trim]     = name_trim(doc[:name_idx])
+      doc[:name_trim_rev] = name_trim(doc[:name_rev])
+
+      make_alt_names(doc) if doc[:alt_names].present?
+    end
+
+    def make_alt_names(doc)
+      doc[:alt_idx] = strip_punct(doc[:alt_names])
+      doc[:alt_idx] = filter_words(doc[:alt_idx], STOPWORDS)
+
+      make_names_with_common(doc, 'alt') unless (doc[:alt_idx].map(&:downcase).join(' ').split & COMMON_WORDS).empty?
+
+      doc[:alt_rev]      = name_rev(doc[:alt_idx])
+      doc[:alt_trim]     = name_trim(doc[:alt_idx])
+      doc[:alt_trim_rev] = name_trim(doc[:alt_rev])
+    end
+
+    def make_names_with_common(doc, prefix)
+      doc[:"#{prefix}_with_common"]          = doc[:"#{prefix}_idx"]
+      doc[:"#{prefix}_rev_with_common"]      = name_rev(doc[:"#{prefix}_with_common"])
+      doc[:"#{prefix}_trim_with_common"]     = name_trim(doc[:"#{prefix}_with_common"])
+      doc[:"#{prefix}_trim_rev_with_common"] = name_trim(doc[:"#{prefix}_rev_with_common"])
+      doc[:"#{prefix}_idx"]                  = filter_words(doc[:"#{prefix}_idx"], COMMON_WORDS)
+    end
+
+    def strip_punct(name)
+      name.class == String ? name.gsub(/[[:punct:]]/, '') : name.map { |n| n.gsub(/[[:punct:]]/, '') }
+    end
+
+    def filter_words(name, wordlist)
+      name.class == String ? name.split.delete_if { |n| wordlist.include?(n.downcase) }.join(' ') : name.map { |n| n.split.delete_if { |word| wordlist.include?(word.downcase) }.join(' ') }
+    end
+
+    def name_rev(name)
+      name.class == String ? name.split.reverse.join(' ') : name.map { |n| n.split.reverse.join(' ') }
+    end
+
+    def name_trim(name)
+      name.class == String ? name.gsub(/\s+/, '') : name.map { |n| n.gsub(/\s+/, '') }
     end
```

Comment:
Ideally you'd have a spec for this module which showed these methods in action against simple examples. It's especially useful to create such specs when a bug is found, as you're able to show "before" (failing) and "after" (passing) behaviors.